### PR TITLE
docs

### PR DESCRIPTION
**Pull request by leoj3n**
WIP, feedback welcome.

### CODE REVIEW

File: tags/function.js [hunk 1]
```diff
@@ -1,120 +1,119 @@
...
-	namedFunction = /\s*function\s+([\w\.\$]+)\s*(~)?\(([^\)]*)/;
-
+var updateNameWithScope = require("../lib/updateNameAndParentWithScope");
+var getParent = require("bit-docs-process-tags/get-parent");
+var tnd = require("bit-docs-type-annotate").typeNameDescription;
+
+// key: function() or key= function(){}
+//(~)? is just a stupid way of making sure there are the right number of parts
+keyFunction = /(?:([\w\.\$]+)|(["'][^"']+["']))\s*[:=].*function\s?\(([^\)]*)/;
+namedFunction = /\s*function\s+([\w\.\$]+)\s*(~)?\(([^\)]*)/;
```
Comment (chasenlehara) [inline tags/function.js:8]: Now that these aren’t comma-separated, they should have a `var` declaration.

**Review by chasenlehara — CHANGES_REQUESTED** [review]
👍

**Comment by chasenlehara** [comment]
In the future, I wouldn’t worry about cleaning up the code. 😊 It’s hard to see what docs were actually changed when there are a bunch of other changes in the code.

**Comment by leoj3n** [comment]
I agree it's hard to see on the overall PR. I tried to make a compromise by doing doc changes and formatting changes in separate commits. I guess that means you have to step through individual commits, which isn't optimal.

What do you think is the best way to handle code that could really use some formatting love, but also needs editing? Kind of a chicken/egg problem. 🐣

We can:

1. Wait for edits to get merged with bad formatting before reformatting...
2. Wait for reformatting to get merged before making edits...
3. Have both in one PR but on separate commits like they are here.

Being able to make edits is most important, so wouldn't want to be waiting on formatting long.

**Comment by leoj3n** [comment]
Reading your comment again @chasenlehara — I guess you're saying don't reformat the code at all.

I'm cool with that. I think I was paranoid the docs weren't being built correctly at one point due to the crazy formatting with mixed tabs/spaces in front of comment blocks, and I reformatted because of that (but that wasn't the issue after all). It was also harder to follow the code before reformatting, but I certainly could have got by without touching the formatting.

For these files it might be a bit of a hassle for me to undo the formatting, so hopefully can keep them how they are in this PR, but I'll try not to touch any formatting in other files.

**Comment by chasenlehara** [comment]
I would wait to reformat stuff until we add ESLint to our projects, but if you come across stuff that’s so bad it’s unreadable, go ahead and fix it and submit it as a PR. You can still branch off a formatting branch to make documentation changes and submit those as a separate PR.

**Comment by leoj3n** [comment]
Okay, that makes sense.

**Review by chasenlehara — APPROVED** [review]

**Pull request merged**